### PR TITLE
fix(unhead): support templateParams in `link` tags

### PR DESCRIPTION
**Pull request by danbars**
Added support for templateParams plugin in link tags. E.g.

```
useHead({
      templateParams: {
        site: {
          name: 'My Site',
          url: 'https://example.com',
        },
        separator: '-',
        path: '/some/page',
      },
      link: [
        {
          rel: 'canonical',
          href: '%site.url%path',
        },
      ],
    })
```

### CODE REVIEW

**Comment by danbars** [comment]
Also allow overriding `templateParams` in pages.
Right now, because dedup runs first, it removes templateParams before it has a chance to affect tags that were defined in a higher level pace (like the app root or the layout).

For example, if this is defined at the app's root:

```
useHead({
      templateParams: {
        site: {
          name: 'My Site',
          url: 'https://example.com',
        },
        separator: '-',
        path: '',
      },
     
      link: [
        {
          rel: 'canonical',
          href: '%site.url%path',
        },
      ],
    })
```
and on a page I have this:
```
useHead({
      templateParams: {
        path: '/some/page',
      },
    })
```

The expected outcome is
```
<link rel="canonical" href="https://example.com/some/path">
```

File: packages/unhead/src/createHead.ts [hunk 1]
```diff
@@ -25,9 +25,9 @@ import { IsBrowser } from './env'
 
 export const CorePlugins = () => [
   // dedupe needs to come first
+  TemplateParamsPlugin(),
```
Comment (harlan-zw) [inline packages/unhead/src/createHead.ts:28]: Should be safe to revert this, the deduping was fixed with https://github.com/unjs/unhead/commit/f9bd644d125d7319298ee8028e5f9936c11f9d71

**Review by harlan-zw — COMMENTED** [review]
Hey @danbars, thanks for the PR!

Happy to support this, I've left one comment

**Comment by danbars** [comment]
Hi @harlan-zw , I've removed the inheritance part since you've already added support for it.
Thanks!

**Pull request merged**